cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -44,6 +44,7 @@ var cloudproviders = []string{
 	"aws",
 	"azure",
 	"cloudstack",
+	"external",
```

Comment:
I haven't considered TLS bootstrapping, I thought [--apiserver-cert-extra-sans](https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/cmd/init.go#L134-L137) would address the problem at first but I'm not sure anymore. I'm not familiar with kubeadm enough to say, I'll do some digging and get back to you. 

Given external cloud controller manager is alpha it seemed reasonable to support external cloud providers and see where it'll go from there. I wouldn't be surprised if it broke for people attempting to use kubeadm with `--cloud-provider=external` but I figured it'll be difficult to catch bugs and gather feedback without supporting the flag to begin with.